src/gui/DatabaseWidget.cpp

```diff
@@ -333,6 +334,18 @@ void DatabaseWidget::cloneEntry()
     return;
 }
 
+void DatabaseWidget::getTotp()
+{
+    Entry* currentEntry = m_entryView->currentEntry();
+    if (!currentEntry) {
+        Q_ASSERT(false);
+        return;
+    }
+
+    TotpDialog* totpDialog = new TotpDialog(this, currentEntry);
```

Comment:
And `this` refers to a DatabaseWidget, that probably is a long lived object.
Possible fix: `totpDialog->setAttribute(Qt::WA_DeleteOnClose);`